BLUESPAWN-client/src/monitor/Event.cpp

```diff
@@ -1,44 +1,182 @@
 #include "monitor/Event.h"
 #include "hunt/reaction/Log.h"
+#include "util/eventlogs/EventLogs.h"
 
 Event::Event(EventType type) : type(type) {}
 
-void Event::setReaction(const Reaction& react) {
-	this->reaction = react;
+void Event::AddCallback(const std::function<void()>& callback) {
+	callbacks.push_back(callback);
 }
 
-void Event::setScope(const Scope& scope) {
-	this->scope = std::optional<Scope>{ scope };
-}
-
-void Event::addCallback(std::function<void(const Scope & scope, Reaction reaction)> callback) {
-	this->callbacks.push_back(callback);
+void Event::RunCallbacks() const {
+	for(auto callback : callbacks){
+		callback();
+	}
 }
 
 /************************
 ***   EventLogEvent   ***
 *************************/
-EventLogEvent::EventLogEvent(std::wstring channel, int eventID, std::vector<EventLogs::XpathQuery> queries) :
-	Event(EventType::EventLog) , channel(channel), eventID(eventID), queries(queries) {
-	eventLogTrigger = std::bind(&EventLogEvent::eventLogCallback, this, std::placeholders::_1);
-}
+EventLogEvent::EventLogEvent(const std::wstring& channel, int eventID, const std::vector<EventLogs::XpathQuery>& queries) :
+	Event(EventType::EventLog), 
+    channel(channel), 
+	eventID(eventID), 
+	eventLogTrigger{ [this](EventLogs::EventLogItem){ this->RunCallbacks(); } } {}
 
-void EventLogEvent::eventLogCallback(EventLogs::EventLogItem item) {
-	for (auto callback : this->callbacks)
-		callback(Scope(), Reactions::LogReaction());
+bool EventLogEvent::Subscribe(){
+	DWORD status{};
+	auto subscription = EventLogs::SubscribeToEvent(GetChannel(), GetEventID(), eventLogTrigger, queries);
+	if(subscription){
+		eventSub = *subscription;
+	}
+	return status;
 }
 
-void EventLogEvent::setEventSub(std::unique_ptr< EventSubscription> sub) {
-	this->eventSub = std::move(sub);
-}
-
-std::wstring EventLogEvent::getChannel() {
+std::wstring EventLogEvent::GetChannel() const {
 	return channel;
 }
-int EventLogEvent::getEventID() {
+
+int EventLogEvent::GetEventID() const {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'thread' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5011005)